_posts/2020-11-07-for-loops-how-to.md

```diff
@@ -11,38 +11,46 @@ are important as a bioinformatics tool.
+the same action on multiple different potential inputs. This is an example of a basic function
 
-Jekyll requires blog post files to be named according to the following format:
+```
+square_this <- function(x) {
+  if (!is.numeric(x)) {
+    stop("ERROR: You did not use a numeric value.")
+  }
+  x^2
+}
+```
+This function is designed to take a numeric input (if it is a non numeric input the function will not run) and add two to it. It then stores this function into the variable "square_this" in order to call the function for use at a later time.  
 
-`YEAR-MONTH-DAY-title.MARKUP`
+# What is a for loop?
+Using a function makes your code more flexible. But say you had numerous different inputs you wanted to run. Running the function for each input one at a time can be tedious. If you had hundreds of different inputs to run inputting each and every one would be impractical. This is where for loops come in.
 
-Where `YEAR` is a four-digit number, `MONTH` and `DAY` are both two-digit numbers, and `MARKUP` is the file extension representing the format used in the file. After that, include the necessary front matter. Take a look at the source for this post to get an idea about how it works.
+A for loop is a special kind of command that allows you to run the same function as many times as you wish but substituting the input file each time. This makes your code even more flexible as it allows you to run the same code many times over in a fraction of the time it would normally take to manually input each and every input file. Let us take a look at a relatively simple for loop.
 
-Jekyll also offers powerful support for code snippets:
+# Anatomy of a for loop
 
-```ruby
-def print_hi(name)
-  puts "Hi, #{name}"
-end
-print_hi('Tom')
-#=> prints 'Hi, Tom' to STDOUT.
+Let's take a look at a basic for loop:
 ```
+for (my_number in 1:5) {
+  my_number <- square_this(my_number)
+  my_number <- add_two_to_this(my_number)
+  print(my_number)
+}
 
-If you want to add an image, you will need to add it to the assets/ directory, and then use the following special syntax where you want to include the image.
-
-Note that the curly braces and site URL part needs to stay as is, the part you change is the image name and the alternative text between the square brackets.
-
-```
-![useful image alternative text]({% raw %}{{ site.url }}{% endraw %}/assets/the-name-of-your-image.png)
+my_number = {"1", "2", "3", "4", "5"}
 ```
+Before doing anything else make sure your function is called into your global environment.
+
+To start out the for loop type for (). This will start out the for loop. In the parentheses place the name of the variables that you want to start in the loop and when you want to finish. In this case we want the the values from 1 through 5. Each of these variables will be represented by "my_number" whenever the for loop re-runs itself.
```

Comment:
Small grammar fix: "the the" on line 46.

Reply:
Also may want to consider putting quotes or backticks `` around what you are instructing your reader to type.

Reply:
Good catch on the grammar! I like the backticks recommendation!